nucypher/utilities/sandbox/ursula.py

```diff
@@ -0,0 +1,166 @@
+import random
+from typing import Set
+
+from constant_sorrow import constants
+from eth_utils import to_checksum_address
+from twisted.internet import protocol
+
+from nucypher.characters.lawful import Ursula
+from nucypher.crypto.api import secure_random
+from nucypher.utilities.sandbox.constants import (DEFAULT_NUMBER_OF_URSULAS_IN_DEVELOPMENT_NETWORK,
+                                                  TEST_URSULA_STARTING_PORT,
+                                                  TEST_KNOWN_URSULAS_CACHE)
+
+
+def make_federated_ursulas(quantity=DEFAULT_NUMBER_OF_URSULAS_IN_DEVELOPMENT_NETWORK,
+                           know_each_other=True,
+                           **ursula_kwargs) -> Set[Ursula]:
+
+    if not TEST_KNOWN_URSULAS_CACHE:
+        starting_port = TEST_URSULA_STARTING_PORT
+    else:
+        starting_port = max(TEST_KNOWN_URSULAS_CACHE.keys()) + 1
+
+    federated_ursulas = set()
+    for port in range(starting_port, starting_port+quantity):
+
+        ursula = Ursula(is_me=True,
+                        db_name="test-{}".format(port),
+                        rest_host="localhost",
+                        rest_port=port + 100,
```

Comment:
If we don't have a DHT anymore, why `+ 100`?

Reply:
Hmm, good point. I'll change it.

Reply:
Changed upstream (to avoid additional rebases).